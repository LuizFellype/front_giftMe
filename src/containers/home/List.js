import React from 'react'
import PropTypes from 'prop-types'
import { DataTable } from 'primereact/datatable'
import { Column } from 'primereact/column'

// eslint-disable-next-line react/display-name
export const ProductList = React.memo(({ products = [] }) => (
  <div className='table-block'>
    <DataTable value={products} autoLayout responsive>
      <Column field='productName' header='Nome do produto' sortable />
      <Column field='url' header='Loja/Lugar' sortable />
    </DataTable>
  </div>
))

ProductList.propTypes = {
  products: PropTypes.array.isRequired
}
