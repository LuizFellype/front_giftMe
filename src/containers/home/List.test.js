import React from 'react'
import { render, cleanup } from 'react-testing-library'
import { ProductList } from './List'

afterEach(cleanup)

describe('Product list render withou crash', () => {
  test('Check html sctructure of the component', () => {
    const { asFragment } = render(<ProductList products={[]} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Product List has wrapper', () => {
    const { getByTestId } = render(<ProductList products={[]} />)
    expect(getByTestId('table-wrapper').className).toContain('table-block')
  })
})
