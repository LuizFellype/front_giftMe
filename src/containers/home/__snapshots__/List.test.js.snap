// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list render withou crash Check html sctructure of the component 1`] = `
<DocumentFragment>
  <div
    class="table-block"
    data-testid="table-wrapper"
  >
    <div
      class="p-datatable p-component p-datatable-responsive p-datatable-auto-layout"
    >
      <div
        class="p-datatable-wrapper"
      >
        <table>
          <thead
            class="p-datatable-thead"
          >
            <tr>
              <th
                class="p-sortable-column"
                tabindex="0"
              >
                <span
                  class="p-column-title"
                >
                  Nome do produto
                </span>
                <span
                  class="p-sortable-column-icon pi pi-fw pi-sort"
                />
              </th>
              <th
                class="p-sortable-column"
                tabindex="0"
              >
                <span
                  class="p-column-title"
                >
                  Loja/Lugar
                </span>
                <span
                  class="p-sortable-column-icon pi pi-fw pi-sort"
                />
              </th>
            </tr>
          </thead>
          <tbody
            class="p-datatable-tbody"
          >
            <tr
              class="p-datatable-emptymessage"
            >
              <td
                colspan="2"
              >
                No records found
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
